packages/react/src/components/AccountSettings/ChangePassword/ChangePassword.tsx

```diff
@@ -1,53 +1,168 @@
 import React from 'react';
+import { isEqual } from 'lodash';
 
 import { Logger } from 'aws-amplify';
-import { changePassword, translate } from '@aws-amplify/ui';
+import {
+  changePassword,
+  FieldValidator,
+  FormEventType,
+  getConfirmPasswordValidator,
+  getDefaultPasswordValidators,
+  runFieldValidators,
+  translate,
+} from '@aws-amplify/ui';
 
 import { useAuth } from '../../../internal';
 import { View, Flex } from '../../../primitives';
+import { FormValues, BlurredFields, ValidationError } from '../types';
 import {
+  DefaultConfirmPassword,
   DefaultCurrentPassword,
   DefaultError,
   DefaultNewPassword,
   DefaultSubmitButton,
 } from './defaultComponents';
 import { ChangePasswordProps } from './types';
-import { FormValues } from '../types';
 
 const logger = new Logger('ChangePassword');
 
+const getIsDisabled = (
+  formValues: FormValues,
+  validationError: ValidationError
+): boolean => {
+  const { currentPassword, newPassword, confirmPassword } = formValues;
+
+  if (!currentPassword || !newPassword || !confirmPassword) {
+    // if passwords aren't entered yet, disable submit
+    return true;
+  } else if (
+    // if there are some password validation error, disable submit
+    validationError.newPassword?.length > 0 ||
+    validationError.confirmPassword?.length > 0
+  ) {
+    return true;
+  }
+
+  return false;
```

Comment:
What do you think if we make it more self-explained?

```suggestion
const noPasswordEntered = !currentPassword || !newPassword || !confirmPassword;
const invalidPassword = validationError.newPassword?.length > 0 || validationError.confirmPassword?.length > 0;

return noPasswordEntered ||  invalidPassword;
```

Reply:
I love this suggestion, so much cleaner. Done in [52d5b02](https://github.com/aws-amplify/amplify-ui/pull/2900/commits/52d5b021b566d63b98ed4a4d1573aa2b98f468ed)!